engineering/documentation.md

```diff
@@ -0,0 +1,226 @@
+projects should provide.
+
+## General Best Practices
+
+People read documentation in order to achieve a goal.
+Whenever you write a new piece you should keep their goal in mind.
+
+For instance, you're currently reading this document.
+What is your goal?
+Maybe you want to improve the documentation of your own project,
+or perhaps you've been told this is an awesome document and your goal
+is simply to enjoy reading a beautiful text of astonishing beauty.
+
+On the other hand I know you're a developer, so there's no point explaining
+what source code is. That would seem patronizing or a waste of time.
+
+I also know that most of you don't speak English as a first language.
+Unnecessarily complicated words or sentences are then an obstacle to our communication.
+
+Finally, I know you're a diverse group. Among the readers of this document you'll find people
+everywhere in the gender spectrum, different races, and different sensibilities.
+I do not wish to alienate any of them, so I need to be careful with the words I choose.
+
+All of these concrete examples can be boiled down to the following best practices.
+
+### State a clear goal
+
+A piece of documentation has a clearly stated goal and audience.
+
+For instance, a `README.md` should be an overview of the project, maybe
+including some basic usage instructions. On the other hand `CONTRIBUTING.md`
+should cover what contributors care about.
+
+Do not make a first time visitor read through how to contribute a project before
+they understand what the project is about.
+
+### Define a scope
+
+As Carl Sagan [famously said](https://www.youtube.com/watch?v=7s664NsLeFM)
+"If you wish to make an apple pie from scratch, you must first invent the universe".
+Would you start an apple pie recipe with quantum physics? Hopefully not.
+
+Same goes for documentation. You need to define the scope of concepts that deserve
+to be explained in a document. Those considered either too advanced or as prerequisites
+should be ommited. You can always link to some other document explaining them.
+
+For instance, if a project uses Go and gRPC feel free to link to the homepages
+of those projects. Do not use more than one short sentence to explain what Go
+or gRPC are.
+
+### Keep it simple
+
+Short sentences win. Avoid having overly complex sentences.
+It might seem weird at first. You might find the resulting text immature.
+But keeping your sentences short help the readers, and that should trump everything else.
+
+Following the same logic, avoid overly complicated words.
+
+As a rule of thumb, sentences longer than a printed line are probably too long.
+
+### Don't alienate the reader
+
+Avoid gendered pronouns. "He or she" should be rewritten as "they".
+There is no reason for your document to make assumptions on the reader's identity.
+
+Avoid sexist connotations.
+For instance, "mom-proof" implies mothers are not technical. Tell that to Grace Hopper.
```

Comment:
Actually Grace Hopper never had children. Let's try also not to assume any woman is a mom ;)

Reply:
good point! replaced mothers with women
thanks :)